tests/metrics/test_ddp.py

```diff
@@ -43,3 +44,30 @@ def _test_ddp_sum_cat(rank, worldsize):
 @pytest.mark.parametrize("process", [_test_ddp_cat, _test_ddp_sum, _test_ddp_sum_cat])
 def test_ddp(process):
     torch.multiprocessing.spawn(process, args=(2,), nprocs=2)
+
+
+def _test_memory_warning(rank, worldsize):
+    setup_ddp(rank, worldsize)
+
+    class DummyMetric(Metric):
+        def __init__(self):
+            super().__init__()
+            self.add_state("x", default=[], dist_reduce_fx=None)
+
+        def update(self, x):
+            self.x.append(x)
+
+        def compute(self):
+            x = torch.cat(self.x, dim=0)
+            return x.sum()
+
+    metric = DummyMetric()
+    metric.update(torch.randn(10, 5)[:, 0])
+
+    with pytest.warns(UserWarning, match="Syncing with `gather_all` requires input to be contiguous"):
+        val = metric.compute()
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="DDP not available on windows")
```

Comment:
of course if we remove the condition and the warning above, this test is no longer valid, instead we could check it to just run.